contracts/Strategy.sol

```diff
@@ -90,17 +90,29 @@ contract Strategy is AccessControl, ERC20Rewards, StrategyMigrator { // TODO: I'
         state = target;
     }
 
+    /// @dev State and state variable management
+    /// @param target State to transition to
+    function _transition(State target) internal {
+        require (target != State.INVESTED, "Must provide a pool");
+        _transition(target, IPool(address(0)));
+    }
+
     // ----------------------- INVEST & DIVEST --------------------------- //
 
-    /// @dev Mock pool mint hooked up to initialize the strategy and return strategy tokens.
+    /// @notice Mock pool mint called by a strategy when trying to migrate.
+    /// @dev Will initialize the strategy and return strategy tokens.
+    /// It is expected that base has been transferred in, but no fyTokens
+    /// @return baseIn Amount of base tokens found in contract
+    /// @return fyTokenIn This is always returned as 0 since they aren't used
+    /// @return minted Amount of strategy tokens minted from base tokens which is the same as baseIn
     function mint(address, address, uint256, uint256)
         external
         override
         auth
```

Comment:
it's a bit duplicitous, but we did lose the `isState(State.DEPLOYED)` check by using an internal fn here.   Prob wouldn't hurt to add it back but not a big deal either way tbh.